plugins/modules/proxmox.py

```diff
@@ -497,6 +503,10 @@ def create_instance(self, vmid, node, disk, storage, cpus, memory, swap, timeout
                     self.module.fail_json(msg='%s is not a valid tag' % tag)
             kwargs['tags'] = ",".join(kwargs['tags'])
 
+        if 'ostype' in kwargs:
+            if kwargs['ostype'] == 'auto':
+                kwargs.pop('ostype')
```

Comment:
Why not:
```suggestion
        if kwargs.get('ostype') == 'auto':
            kwargs.pop('ostype')
```

Reply:
Sorry, the code I wrote out of habit remained 🙇